server/jetstream_cluster.go

```diff
@@ -7193,7 +7214,6 @@ func (mset *stream) runCatchup(sendSubject string, sreq *streamSyncRequest) {
 	// Setup sequences to walk through.
 	seq, last := sreq.FirstSeq, sreq.LastSeq
 	mset.setCatchupPeer(sreq.Peer, last-seq)
-	defer mset.clearCatchupPeer(sreq.Peer)
```

Comment:
What other ways can we exit without clearing? We ok with those?

Reply:
These are the only return statements after the removed line.
We do not clear, when there is no quit channel, quit channel or server quit channel gets closed.

We do not clear it when the other side aborts catchup and closes remoteQuitCh.
(log msg: Catchup for stream '%s > %s' was aborted on the remote due to: %q")
@derekcollison For this situation I'm leaning towards not clearing, unless you say this actually signals success. 

Basically we only clear when this side determines catchup is finished.
As a fallback to not clearing catchup, when a server looses leadership it clears all of them. 

```
	if qch == nil {
		return
	}
  ...
	for {
		select {
		case <-s.quitCh:
			return
		case <-qch:
			return
		case <-remoteQuitCh:
			return
		case <-notActive.C:
			s.Warnf("Catchup for stream '%s > %s' stalled", mset.account(), mset.name())
			return
		case <-nextBatchC:
			if !sendNextBatchAndContinue(qch) {
				mset.clearCatchupPeer(sreq.Peer)
				return
			}
		case <-cbKick:
			if !sendNextBatchAndContinue(qch) {
				mset.clearCatchupPeer(sreq.Peer)
				return
			}
		}
	}
}
```

Reply:
Would need to look closer, but I would think maybe clear on that..

Reply:
fixed